stumpy/stump.py

```diff
@@ -201,14 +201,13 @@ def _compute_diagonal(
 
             if T_B_subseq_isfinite[uint64_j] and T_A_subseq_isfinite[uint64_i]:
                 # Neither subsequence contains NaNs
-                if T_B_subseq_isconstant[uint64_j] or T_A_subseq_isconstant[uint64_i]:
+                if T_B_subseq_isconstant[uint64_j] and T_A_subseq_isconstant[uint64_i]:
```

Comment:
My guess:

Maybe it had something to do with the loss of precision in stddev. Previously, we used stddev threshold instead of `isconstant`. If we have two subseqs with very small `stddev`, their z-norm distance is not necessarily zero....but we wanted to assume that it was zero. So, previously, we had this final if block:

```
 if stddev_B<threshold and stddev_A<threshold:
     pearson = 1.0
```
to reset pearson to 1.0 if the stddev of both subseqs are very small. Now, we are trying to be precise in that regard. So, we should be fine now.

Reply:
My guess:

Maybe it had something to do with the loss of precision in stddev. Previously, we used stddev threshold instead of `isconstant`. If we have two subseqs with very small `stddev`, their z-norm distance is not necessarily zero....but we wanted to assume that it was zero. So, previously, we had this final if block:

```
 if stddev_B<threshold and stddev_A<threshold:
     pearson = 1.0
```
to reset pearson to 1.0 (i.e. their distance is 0) if the stddev of both subseqs are very small. Now, we are trying to be precise in that regard. So, we should be fine now.